### PR TITLE
PS-2246 Configurable workspace backend

### PR DESCRIPTION
**Pull request by joker806**
https://keboola.atlassian.net/browse/PS-2246
Depends on https://github.com/keboola/job-queue-internal-api/pull/64

### CODE REVIEW

File: src/JobFactory/FullJobDefinition.php [hunk 1]
```diff
@@ -114,6 +114,9 @@ protected function getRootDefinition(TreeBuilder $treeBuilder): ArrayNodeDefinit
                         ->end()
                     ->end()
                 ->end()
+                ->arrayNode('backend')->ignoreExtraKeys(true)->children()
+                    ->scalarNode('type')->end()
+                ->end()->end()
```
Comment (odinuv) [inline src/JobFactory/FullJobDefinition.php:119]: indent

File: src/JobFactory/NewJobDefinition.php [hunk 2]
```diff
@@ -62,6 +62,9 @@ protected function getRootDefinition(TreeBuilder $treeBuilder): ArrayNodeDefinit
                         ->end()
                     ->end()
                 ->end()
+                ->arrayNode('backend')->ignoreExtraKeys(false)->children()
+                    ->scalarNode('type')->end()
+                ->end()->end()
```
Comment (odinuv) [inline src/JobFactory/NewJobDefinition.php:67]: indent

**Review by odinuv — APPROVED** [review]
seems legit

File: src/JobFactory/FullJobDefinition.php (hunk 1 above)
Comment (joker806) [reply src/JobFactory/FullJobDefinition.php]: Takto?

File: src/JobFactory/NewJobDefinition.php (hunk 2 above)
Comment (joker806) [reply src/JobFactory/NewJobDefinition.php]: Takto?

File: src/JobFactory/FullJobDefinition.php (hunk 1 above)
Comment (odinuv) [reply src/JobFactory/FullJobDefinition.php]: jj

**Pull request merged**